# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1252498472 -7200
#      Wed Sep 09 14:14:32 2009 +0200
# Branch release-3-2-x
# Node ID 6ba99f799ec1ea3f2c413e38df2b2ed16b05df65
# Parent  330cf1e13fb617058094814209d0422d28eaab9e
pr-output.cc: new test

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-09-08  Rafael Laboissiere  <rafael@debian.org>
+
+	* pr-output.cc: New test.
+
 2009-09-03  John W. Eaton  <jwe@octave.org>
 
 	* pr-output.cc (set_format (const Complex&, int&, int&)):
 	Avoid passing NaN or Inf to log10.
 
 2008-09-01  David Bateman  <dbateman@free.fr>
 
 	* DLD-FUNCTIONS/eig.cc (Feigs): Correct nesting error in option
diff --git a/src/pr-output.cc b/src/pr-output.cc
--- a/src/pr-output.cc
+++ b/src/pr-output.cc
@@ -3345,16 +3345,28 @@ Note that the output from @code{fdisp} a
 	}
     }
   else
     print_usage ();
 
   return retval;
 }
 
+/* 
+%!test
+%! format short
+%! fd = tmpfile ();
+%! for r = [0, Inf -Inf, NaN]
+%!   for i = [0, Inf -Inf, NaN]
+%!     fdisp (fd, complex (r, i));
+%!   endfor
+%! endfor
+%! fclose (fd);
+*/                                 
+
 static void
 init_format_state (void)
 {
   free_format = false;
   plus_format = false;
   rat_format = false;
   bank_format = false;
   hex_format = 0;
